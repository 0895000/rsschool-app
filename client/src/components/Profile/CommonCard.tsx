import * as React from 'react';
import {
  Card,
  Typography,
} from 'antd';

const { Title } = Typography;

type Props = {
  title: string;
  icon: any;
  content: any;
  actions?: any;
};

class CommonCard extends React.Component<Props> {
  render() {
    const { title, icon, content, actions } = this.props;
    return (
      <Card
        title={
          <Title
            level={2}
            ellipsis={true}
            style={{ fontSize: 16, marginBottom: 0 }}
          >
            {icon} {title}
          </Title>
        }
        actions={actions ? actions : []}
      >
        {content}
      </Card>
    );
  }
}

export default CommonCard;
