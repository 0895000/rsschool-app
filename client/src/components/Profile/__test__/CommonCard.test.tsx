import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import CommonCard from '../CommonCard';

describe('CommonCard', () => {
  describe('Should render correctly', () => {
    it('if just basic props is present', () => {
      const output = shallow(
        <CommonCard
          title="Test"
          icon={<i>Icon</i>}
          content={<p>Card body</p>}
        />,
      );
      expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('if is editing mode enabled', () => {
      const output = shallow(
        <CommonCard
          title="Test"
          icon={<i>Icon</i>}
          content={<p>Card body</p>}
          isEditingModeEnabled={true}
          profileSettingsContent={<p>Settings</p>}
        />,
      );
      expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('if is null content passed', () => {
      const output = shallow(
        <CommonCard
          title="Test"
          icon={<i>Icon</i>}
          content={null}
        />,
      );
      expect(shallowToJson(output)).toMatchSnapshot();
    });
  });

  const wrapper = shallow<CommonCard>(
    <CommonCard
      title="Test"
      icon={<i>Icon</i>}
      content={<p>Content</p>}
    />);
  const instance: any = wrapper.instance();
  describe('showVisibilitySettings', () => {
    it('should set "state.isVisibilitySettingsVisible" as "true"', () => {
      expect(instance.state.isVisibilitySettingsVisible).toBe(false);
      instance.showVisibilitySettings();
      expect(instance.state.isVisibilitySettingsVisible).toBe(true);
    });
  });
  describe('hideVisibilitySettings', () => {
    it('should set "state.isVisibilitySettingsVisible" as "false"', () => {
      instance.state.isVisibilitySettingsVisible = true;
      expect(instance.state.isVisibilitySettingsVisible).toBe(true);
      instance.hideVisibilitySettings();
      expect(instance.state.isVisibilitySettingsVisible).toBe(false);
    });
  });
  describe('showProfileSettings', () => {
    it('should set "state.isProfileSettingsVisible" as "true"', () => {
      expect(instance.state.isProfileSettingsVisible).toBe(false);
      instance.showProfileSettings();
      expect(instance.state.isProfileSettingsVisible).toBe(true);
    });
  });
  describe('hideProfileSettings', () => {
    it('should set "state.isProfileSettingsVisible" as "false"', () => {
      instance.state.isProfileSettingsVisible = true;
      expect(instance.state.isProfileSettingsVisible).toBe(true);
      instance.hideProfileSettings();
      expect(instance.state.isProfileSettingsVisible).toBe(false);
    });
  });
});
