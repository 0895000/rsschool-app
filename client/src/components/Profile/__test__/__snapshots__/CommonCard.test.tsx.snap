// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonCard Should render correctly if is editing mode enabled 1`] = `
<Card
  actions={
    Array [
      <ForwardRef(EditOutlined)
        onClick={[Function]}
      />,
    ]
  }
  title={
    <Title
      ellipsis={true}
      level={2}
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 0,
        }
      }
    >
      <i>
        Icon
      </i>
       
      Test
    </Title>
  }
>
  <p>
    Card body
  </p>
  <PermissionsSettingsDrawer
    hideSettings={[Function]}
    isSettingsVisible={false}
  />
  <ProfileSettingsDrawer
    content={
      <p>
        Settings
      </p>
    }
    hideSettings={[Function]}
    isSettingsVisible={false}
  />
</Card>
`;

exports[`CommonCard Should render correctly if is null content passed 1`] = `
<Card
  title={
    <Title
      ellipsis={true}
      level={2}
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 0,
        }
      }
    >
      <i>
        Icon
      </i>
       
      Test
    </Title>
  }
>
  <Empty
    image={<Simple />}
  />
</Card>
`;

exports[`CommonCard Should render correctly if just basic props is present 1`] = `
<Card
  title={
    <Title
      ellipsis={true}
      level={2}
      style={
        Object {
          "fontSize": 16,
          "marginBottom": 0,
        }
      }
    >
      <i>
        Icon
      </i>
       
      Test
    </Title>
  }
>
  <p>
    Card body
  </p>
</Card>
`;
